File: Top-Interview-Questions-Leetcode/Easy-Collections/strings/reverse-string.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    void reverseString(std::vector<char>& s) {
        int l = 0, h = s.size() - 1;
        
        while (l < h) {
            std::swap(s[l++], s[h--]);
        }
    }
};

// Example usage
int main() {
    Solution solution;
    std::vector<char> s = {'h', 'e', 'l', 'l', 'o'};
    
    solution.reverseString(s);
    
    for (char c : s) {
        std::cout << c << " ";
    }
    
    return 0;
}

File: Top-Interview-Questions-Leetcode/Easy-Collections/strings/reverse-string.py
from typing import List

class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        
        l, h = 0, len(s) - 1
        
        while l < h:
            temp = s[l]
            s[l] = s[h]
            s[h] = temp
            
            l += 1
            h -= 1

# Example usage
def main():
    s = ["h", "e", "l", "l", "o"]
    solution = Solution()
    solution.reverseString(s)
    print(s)

if __name__ == "__main__":
    main()